mlserver/parallel/registry.py

```diff
@@ -145,8 +174,18 @@ def _get_env_path(self, env_hash: str) -> str:
 
     async def _find(self, model: MLModel) -> InferencePool:
         env_hash = _get_environment_hash(model)
+        inference_pool_gid = (
+            model.settings.parameters.inference_pool_gid
+            if model.settings.parameters
+            else None
+        )
+
         if not env_hash:
-            return self._default_pool
+            return (
```

Comment:
nit (style): I think it might be more legible to have if condition and multiple return as opposed to inline if-return.